src/ILCompiler.CppCodeGen/src/CppCodeGen/CppWriter.cs

```diff
@@ -1217,6 +1239,37 @@ private void OutputExternCSignatures()
             Out.Write(sb.ToString());
         }
 
+        /// <summary>
+        /// Output C++ code for a given unboxingStubNode
+        /// </summary>
+        /// <param name="unboxingStubNode">The unboxing node to be output</param>
+        /// <param name="methodImplementations">The buffer in which to write out the C++ code</param>
+        private void OutputUnboxNode(UnboxingStubNode unboxingStubNode)
```

Comment:
I was thinking of using a derived type, we need to differentiate at line 851, unless all uses of `GetCppMethodName` are replaced by `method.GetMangledName`